topology/src/main/java/io/camunda/zeebe/topology/api/TopologyManagementRequestsHandler.java

```diff
@@ -83,4 +115,74 @@ private ActorFuture<TopologyChangeStatus> applyOperations(
 
     return responseFuture;
   }
+
+  private List<TopologyChangeOperation> generatePartitionDistributionOperations(
+      final ClusterTopology currentTopology, final Set<MemberId> brokers) {
```

Comment:
:x: Let's validate that `brokers` is not empty, either here or in the constructor for `ReassignPartitionsRequest`. Once this is wired up to an actuator, it could be extremely easy to accidentally reassign to an empty set. In the current state this would result in leave operations for all members and ultimately a lost partition.

(Reminds me, at some point I flagged that a broker should refuse to leave if it is the last remaining. I can't find if/where this is implemented.)

Reply:
:+1: Good point. 

> (Reminds me, at some point I flagged that a broker should refuse to leave if it is the last remaining. I can't find if/where this is implemented.)

Haven't forgotten this, I had added it to the followup tasks in the main epic.